addons/delivery/models/delivery_carrier.py

```diff
@@ -277,3 +278,111 @@ def fixed_get_tracking_link(self, picking):
+        # more in the max weight than in the total weight, so that it only
+        # creates ONE package with everything.
+        max_weight = default_package_type.max_weight or total_weight + 1
+        total_full_packages = int(total_weight / max_weight)
+        last_package_weight = total_weight % max_weight
+
+        package_weights = [max_weight] * total_full_packages + [last_package_weight] if last_package_weight else []
+        partial_cost = total_cost / len(package_weights)  # separate the cost uniformly
+        for weight in package_weights:
+            packages.append(DeliveryPackage(None, weight, default_package_type, total_cost=partial_cost, currency=order.company_id.currency_id))
+        return packages
+
+    def _get_packages_from_picking(self, picking, default_package_type):
+        packages = []
+
+        if picking.is_return_picking:
+            commodities = self._get_commodities_from_stock_move_lines(picking.move_line_ids)
+            weight = picking._get_estimated_weight()
+            packages.append(DeliveryPackage(commodities, weight, default_package_type))
+            return packages
+
+        # Create all packages.
+        for package in picking.package_ids:
+            move_lines = picking.move_line_ids.filtered(lambda ml: ml.result_package_id == package)
+            commodities = self._get_commodities_from_stock_move_lines(move_lines)
+            package_total_cost = 0.0
+            for quant in package.quant_ids:
+                package_total_cost += self._product_price_to_company_currency(quant.quantity, quant.product_id, picking.company_id)
+            packages.append(DeliveryPackage(commodities, package.shipping_weight or package.weight, package.package_type_id, name=package.name, total_cost=package_total_cost, currency=picking.company_id.currency_id))
+
+        # Create one package: either everything is in pack or nothing is.
+        if picking.weight_bulk:
+            commodities = self._get_commodities_from_stock_move_lines(picking.move_line_ids)
+            package_total_cost = 0.0
+            for move_line in picking.move_line_ids:
+                package_total_cost += self._product_price_to_company_currency(move_line.qty_done, move_line.product_id, picking.company_id)
+            packages.append(DeliveryPackage(commodities, picking.weight_bulk, default_package_type, name='Bulk Content', total_cost=package_total_cost, currency=picking.company_id.currency_id))
+
+        return packages
+
+    def _get_commodities_from_order(self, order):
+        commodities = []
+
+        for line in order.order_line.filtered(lambda line: not line.is_delivery and not line.display_type):
+            unit_quantity = line.product_uom._compute_quantity(line.product_uom_qty, line.product_id.uom_id, rounding_method='HALF-UP')
+            rounded_qty = max(1, float_round(unit_quantity, precision_digits=0, rounding_method='HALF-UP'))
```

Comment:
sure it shouldn't be rounded UP?

Reply:
HALF-UP is probably fine

Reply:
`HALF-UP` was used only with easypost, the other ones used the default, which is `UP`. So maybe it would make more sense to use the default for all.. I just wonder why EP was not using it... (will check)